src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -15,10 +15,71 @@
  */
 package nl.knaw.dans.easy.fedora2vault
 
+import java.io.InputStream
+
+import better.files.File
+import com.yourmediashelf.fedora.client.FedoraClient
+import javax.naming.ldap.InitialLdapContext
+import nl.knaw.dans.bag.v0.DansV0Bag
+import nl.knaw.dans.easy.fedora2vault.Command.FeedBackMessage
+import nl.knaw.dans.easy.fedora2vault.FoXml.{ getEmd, _ }
+
 import scala.util.{ Success, Try }
+import scala.xml.{ Elem, XML }
+
+class EasyFedora2vaultApp(configuration: Configuration) {
+  lazy val fedoraProvider: FedoraProvider = new FedoraProvider(new FedoraClient(configuration.fedoraCredentials))
+  lazy val ldapContext: InitialLdapContext = new InitialLdapContext(configuration.ldapEnv, null)
+  private lazy val ldap = new Ldap(ldapContext)
 
-class EasyFedora2vaultApp(configuration: Configuration)  {
+  def simpleTransform(datasetId: DatasetId, outputDir: File): Try[FeedBackMessage] = {
 
+    def managedMetadataStream(foXml: Elem, streamId: String, bag: DansV0Bag, metadataFile: String) = {
+      managedStreamLabel(foXml, streamId)
+        .map { label =>
+          val extension = label.split("[.]").last
+          val bagFile = s"$metadataFile.$extension"
+          fedoraProvider.disseminateDatastream(datasetId, streamId)
+            .map(bag.addMetadataStream(bagFile))
+            .tried.flatten
+        }
+    }
 
+    def compareManifest(bag: DansV0Bag)(streamId: String): Try[Any] = {
+      fedoraProvider.disseminateDatastream(datasetId, streamId)
+        .map { inputStream: InputStream =>
+          val manifestst = bag.payloadManifests
+          Success(()) // TODO EASY-2678
+        }.tried.flatten
+    }
 
+    for {
+      foXml <- fedoraProvider.getObject(datasetId).map(XML.load).tried
+      depositor <- getOwner(foXml)
+      bag <- DansV0Bag.empty(outputDir).map(_.withEasyUserAccount(depositor))
+      _ <- getEmd(foXml)
+        .flatMap(bag.addMetadataXml("emd.xml"))
+      _ <- getAmd(foXml)
+        .flatMap(bag.addMetadataXml("amd.xml"))
+      _ <- getDdm(foXml)
+        .map(bag.addMetadataXml("dataset.xml"))
```

Comment:
note for future improvement (please create an issue for this or include it in an existing issue):
if the `dataset.xml` exists, the metadata contained in it may not be up to date with the dataset's actual metadata due to curations by the datamanager, which are only reflected in EMD. Hence you either might want to (1) compare DDM with EMD and see if their contents are equal or (2) always create a new DDM from the EMD and only use this `dataset.xml` for the parts that are not contained in EMD.